### PR TITLE
Failing test for internal server error

### PR DESCRIPTION
**Pull request by samphilipd**

### CODE REVIEW

File: test/support.exs [hunk 1]
```diff
@@ -24,6 +24,15 @@ defmodule Raxx.Forwarder do
   end
 end
 
+defmodule Raxx.HandleRequestForwarder do
```
Comment (CrowdHailer) [inline test/support.exs]: Probably this should be called `Raxx.RaiseException` or `Raxx.Kaboom!` depending on how you are feeling? curious question can module names also have `!` in them. Also I wouldn't bother with a specific path.

**Comment by CrowdHailer** [comment]
So to confirm the current issue is that this new test fails because no response is sent but the connection stays open

File: test/ace/http1/server_test.exs [hunk 2]
```diff
@@ -56,6 +56,30 @@ defmodule Ace.HTTP1.ServerTest do
...
+    host: example.com
+
+    """
+
+    {:ok, socket} = :gen_tcp.connect({127, 0, 0, 1}, port, [:binary])
+    :ok = :gen_tcp.send(socket, http1_request)
+
+    assert_receive {:tcp, ^socket, response}, 1000
+
+    assert response == "HTTP/1.1 500 Internal server error\r\nconnection: close\r\ncontent-length: 2\r\n"
```
Comment (CrowdHailer) [inline test/ace/http1/server_test.exs]: This response won't have a length of 2

File: test/support.exs (hunk 1 above)
Comment (samphilipd) [reply test/support.exs]: I thought you could match on the path for different behaviours. So then you have a generic HandleRequestForwarder with different paths like `raise_error`, `render_ok` etc

**Comment by samphilipd** [comment]
Yes. The behaviour I would like is a 500 code with a closed connection.

**Comment by CrowdHailer** [comment]
Quick look and I think this is where I need to start
here
https://github.com/CrowdHailer/Ace/blob/master/lib/ace/http/server.ex#L56
and here
https://github.com/CrowdHailer/Ace/blob/master/lib/ace/http/server.ex#L122

The new worker process should be monitored.

File: test/ace/http1/server_test.exs [hunk 3]
```diff
@@ -77,7 +77,7 @@ defmodule Ace.HTTP1.ServerTest do
 
     assert_receive {:tcp, ^socket, response}, 1000
 
-    assert response == "HTTP/1.1 500 Internal server error\r\nconnection: close\r\ncontent-length: 2\r\n"
+    assert response == "HTTP/1.1 500 Internal server error\r\nconnection: close\r\ncontent-length: 21\r\n"
```
Comment (CrowdHailer) [inline test/ace/http1/server_test.exs]: length 21?

Comment (samphilipd) [reply test/ace/http1/server_test.exs]: This may be wrong. Not sure how to calculate it.

Comment (CrowdHailer) [reply test/ace/http1/server_test.exs]: it's the length of the body after the headers so in this case 0.

File: test/support.exs (hunk 1 above)
Comment (CrowdHailer) [reply test/support.exs]: Raxx.Forwarder works for all the render_ok cases etc. Also if going off paths nothing about this module is a forwarder

**Comment by CrowdHailer** [comment]
@samphilipd just a quick heads. if possible please always try and pull the latest from upstream when starting a branch. Was confused with what was going on until I realised this branch was started from a `0.14.x` version. thanks :candy:

**Comment by samphilipd** [comment]
@CrowdHailer Sure, although I was sure I had done so. Let me know next time and I'll rebase on master.

**Pull request merged**